events/sources/kubernetes/kubernetes_source.go

```diff
@@ -78,8 +78,12 @@ type KubernetesEventSource struct {
 
 func (this *KubernetesEventSource) GetNewEvents() *core.EventBatch {
 	startTime := time.Now()
-	defer lastEventTimestamp.Set(float64(time.Now().Unix()))
-	defer scrapEventsDuration.Observe(float64(time.Since(startTime)) / float64(time.Microsecond))
+	defer func() {
```

Comment:
@crassirostris do we want this change?

Reply:
I'm not sure

@serathius Could you please explain it somewhere, e.g. in the description?

Reply:
I was mainly motivated by https://github.com/kubernetes/heapster/pull/1643/files

Reply:
OK, I see, arguments evaluating

This looks ugly anyways, can it be refactored into a single function call?

Reply:
Yep

Reply:
OTOH, let it be, refactoring it seems to add even more complexity and boilerplate